chore(db): use uppercase DB property names

mboskamp opened the pull request:
Related to https://github.com/camunda/camunda-bpm-platform/issues/4525

danielkelemen commented on engine/src/test/java/org/camunda/bpm/engine/test/persistence/DatabaseNamingConsistencyTest.java:
```diff
@@ -0,0 +1,87 @@
+ */
+package org.camunda.bpm.engine.test.persistence;
+
+import org.camunda.bpm.engine.ProcessEngine;
```
🔧 There are some unnecessary imports.

danielkelemen commented on engine/src/test/java/org/camunda/bpm/engine/test/persistence/DatabaseNamingConsistencyTest.java:
```diff
@@ -0,0 +1,87 @@
+
+public class DatabaseNamingConsistencyTest {
+
+    public static String MATCH_REGEX = "([A-Za-z_.]+_)[\\s,]";
+    public static String MATCH_LINE_REGEX = ".*[a-z_]+_[\\s,].*";
```
💬 These won't catch everything. Something like `deployment_ID_` would slip through.
We shouldn't overcomplicate this though, so I think it's fine for the patch but maybe we could have a different regex?

danielkelemen commented on engine/src/test/java/org/camunda/bpm/engine/test/persistence/DatabaseNamingConsistencyTest.java:
```diff
@@ -0,0 +1,87 @@
+                File folder = new File(resource.getFile());
+                File[] filesInFolder = folder.listFiles();
+                for (File mappingFile : filesInFolder) {
+                    Pattern linePattern = Pattern.compile(MATCH_LINE_REGEX);
```
🔧 The patterns could be compiled earlier, outside the loops.

danielkelemen commented on engine/src/test/java/org/camunda/bpm/engine/test/persistence/DatabaseNamingConsistencyTest.java:
```diff
@@ -0,0 +1,87 @@
+        }
+        // then don't expect any results
+        String errorMessage = errorMessageBuilder.toString();
+        if (!"".equals(errorMessage)) {
```
🙃

```suggestion
        if (!errorMessage.isEmpty()) {
```

danielkelemen reviewed (commented):
👍 Looks pretty good, great that you've found more lower case columns!
I added a few comments and thoughts.

mboskamp commented on engine/src/test/java/org/camunda/bpm/engine/test/persistence/DatabaseNamingConsistencyTest.java:
```diff
@@ -0,0 +1,87 @@
+
+    public static String MATCH_REGEX = "([A-Za-z_.]+_)[\\s,]";
+    public static String MATCH_LINE_REGEX = ".*[a-z_]+_[\\s,].*";
+    public static String[] SCANNED_FOLDERS = {"org/camunda/bpm/engine/impl/mapping/entity/"};
```
I will add the liquibase scripts as well as the create and drop scripts. This will ensure that we don't have lowercase characters in table names.
Thanks for the hint!

danielkelemen commented on engine/src/test/java/org/camunda/bpm/engine/test/persistence/DatabaseNamingConsistencyTest.java:
```diff
@@ -0,0 +1,108 @@
+  }
+
+  @Test
+  public void shouldFindLowercaseDbColumnNamesInTestStrings() {
```
👍

danielkelemen reviewed (approved):
👍 Code looks good!

mboskamp commented:
CI failures:
* `quarkus-UNIT`: Expected, fixed with https://github.com/camunda/camunda-bpm-platform/pull/4544
* `ConcurrentTelemetryConfigurationTest` in `old-engine` scenario: Expected, fixed with https://github.com/camunda/camunda-bpm-platform/pull/4544

No more failures, and approved. I will merge it to master now and create the backports.
